src/Tasks/NativeMethods.cs

```diff
@@ -1532,14 +1532,21 @@ private void InitializeEnum(String assemblyName)
                 }
                 else
                 {
-                    if (!string.IsNullOrWhiteSpace(assemblyName))
+                    if (Directory.Exists(s_gacPath))
                     {
-                        _assemblyNameVersion = new AssemblyName(assemblyName);
-                        _gacDirectories = Directory.EnumerateDirectories(s_gacPath, _assemblyNameVersion.Name);
+                        if (!string.IsNullOrWhiteSpace(assemblyName))
+                        {
+                            _assemblyNameVersion = new AssemblyName(assemblyName);
+                            _gacDirectories = Directory.EnumerateDirectories(s_gacPath, _assemblyNameVersion.Name);
+                        }
+                        else
+                        {
+                            _gacDirectories = Directory.EnumerateDirectories(s_gacPath);
+                        }
                     }
                     else
                     {
-                        _gacDirectories = Directory.EnumerateDirectories(s_gacPath);
+                        _gacDirectories = Array.Empty<string>();
```

Comment:
Is it better to set this to `Enumerable.Empty<string>()` or `Array.Empty<string>()`?

Reply:
They're the same: https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Enumerable.cs#L13

Reply:
Method | Mean | Error | StdDev | Allocated
-- | -- | -- | -- | --
EmptyArray | 21.02 ns | 0.0624 ns | 0.0584 ns | 0 B
EmptyEnumerable | 19.25 ns | 0.1355 ns | 0.1268 ns | 0 B

Odd they're moderately different. Maybe it avoids a cast? This for sure doesn't matter, but I ran that real quick before I saw your link.

Reply:
LOL

Reply:
![The more you know](https://media.giphy.com/media/3og0IMJcSI8p6hYQXS/giphy.gif)